### DISCUSSION TITLE
Not working. Continues to charge to 100%

### DISCUSSION THREAD

**Discussion by dmelgar** [AppHouseKitchen/AlDente-Battery_Care_and_Monitoring#313] [General]
I have a MacBook Air M1.
I am running MacOS Monterey 12.0 Beta (21A5248p) - ie beta 1.
I've installed AlDente.
AlDente is running. It appears in the menu bar.
I have it set to 75%
It has (had) been working.
I plugged in the MacBook Air while it was closed, ie clamshelled. 
When I opened the MacBook Air it was at 100%.

Is there some limitation regarding when it works? Does it only work if the MacBook isn't closed?

**Comment by ach1302** [comment]
At first I thought it was not working for me either. I believe my problem is, this is a software solution. I found it I shut it down/power off, then the hardware will charge it to 100%. I was going to delete it until I realized this. Now I unplug the charger when I shut it down.

**Comment by MatthiasKerbl** [comment]
> I have a MacBook Air M1.
> I am running MacOS Monterey 12.0 Beta (21A5248p) - ie beta 1.
> I've installed AlDente.
> AlDente is running. It appears in the menu bar.
> I have it set to 75%
> It has (had) been working.
> I plugged in the MacBook Air while it was closed, ie clamshelled.
> When I opened the MacBook Air it was at 100%.
> 
> Is there some limitation regarding when it works? Does it only work if the MacBook isn't closed?

Hi @dmelgar,

AlDente Free can not control charging while your MacBook is on standby or turned off. Therefore, it will charge to 100% when it is still plugged in.
With AlDente Pro we introduced two more features (Stop charging while sleeping, Stop charging while turned off). With these features activated, your MacBook won't charge while your MacBook is asleep or turned off.

Kind regards,

Matthias

**Comment by Swistunopolis** [comment]
How is that charging stop when sleeping/powered off done technically? I assume via reprogramming the SMC?
This imposes the following problem: when the MacBook sleeps and reaches a very low SoC while in sleep mode, it won't turn on until recharged to a specific value; however, it won't charge if the SMC programming inhibits charging in this mode.
Even worse if charging is disabled also when powered off - then it won't even recharge if it is forcefully powered off, leaving an SMC-reset as the only option to get the system to boot up again!
I have already ran into these issues twice - once with the need for a SMC-reset. This experience has led me to disable the charging inhibitance when powered off, but today I ran into the "SoC drops below bootable limit while in sleep-mode", which was also unpleasent as I had to do an ungraceful hard-shutdown to get the MacBook to accept some charge.

My guess is that you set a charging inhibitor flag in the SMC when entering sleep mode or powering off the machine (or setting these flags when AlDente exits). Maybe you can or should change this behaviour to set a charging limit to e.g. 20% while sleeping or powered off and not preventing charging at all, so that the MacBook can get charged to a level where it can be booted up while in sleep or power-off state.

RFC! :)
